src/lib/ruby-generator/pen.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Define Ruby code generator for Pen Blocks
+ * @param {RubyGenerator} Generator The RubyGenerator
+ * @return {RubyGenerator} same as param.
+ */
+export default function (Generator) {
+    Generator.pen_clear = function () {
+        return 'pen_clear()\n';
+    };
+
+    Generator.pen_stamp = function () {
+        return 'pen_stamp()\n';
+    };
+
+    Generator.pen_penDown = function () {
+        return 'pen_down()\n';
+    };
+
+    Generator.pen_penUp = function () {
+        return 'pen_up()\n';
+    };
+
+    Generator.pen_setPenColorToColor = function (block) {
+        const color = Generator.valueToCode(block, 'COLOR', Generator.ORDER_NONE) || null;
+        return `set_pen_color(${color})\n`;
```

Comment:
setXyzは self.xyz = valueがいいです。

```suggestion
        return `self.pen_color = ${color}\n`;
```